pennylane/hf/tapering.py

```diff
@@ -0,0 +1,277 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the functions needed for tapering qubits using symmetries.
+"""
+from copy import deepcopy
+import numpy as np
+import pennylane as qml
+
+
+def _binary_matrix(terms, num_qubits):
+    r"""Get a binary matrix representation of the hamiltonian where each row coressponds to a
+    Pauli term, which is represented by a concatenation of Z and X vectors.
+
+    Args:
+        terms (Iterable[Observable]): operators defining the Hamiltonian.
```

Comment:
Maybe it is more convenient to just pass the Hamiltonian, instead of its ops?
Also no need to "." at the end.